### PR TITLE
Add Basic Support for Discord

### PR DESCRIPTION
**Pull request by ATechAdventurer**
Added Discord Lookup function and allows the account to be displayed on the profile.
Screenshot 2023-09-16 at 2 43 52 PM

### CODE REVIEW

File: keycloak/keycloak.go [hunk 1]
```diff
@@ -111,6 +112,24 @@ func (k *Keycloak) GetUser(ctx context.Context, userID string) (*User, error) {
 	user.StripeCancelationTime, _ = strconv.ParseInt(safeGetAttr(kcuser, "stripeCancelationTime"), 10, 0)
 	user.StripeETag = safeGetAttr(kcuser, "stripeCancelationTime")
 
+	kuserlogins, err := k.client.GetUserFederatedIdentities(ctx, token.AccessToken, k.env.KeycloakRealm, *kcuser.ID)
+
```
Comment (jveski) [inline keycloak/keycloak.go]: nit: it's idiomatic to not leave an empty line between a stanza and it's error handling code

```suggestion
```

File: discord/discord.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	ID        string `json:"id"`
+	CreatedAt string `json:"created_at"`
+	Username  string `json:"global_name"`
+	Avatar    struct {
+		URL string `json:"link"`
+	} `json:"avatar"`
+}
+
+func GetDiscordUserData(id string) (*DiscordUserData, error) {
+	url := "https://discordlookup.mesavirep.xyz/v1/user/" + id
```
Comment (jveski) [inline discord/discord.go:19]: Nice API 👍

File: discord/discord.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("received non-OK HTTP status: %s", res.Status)
+	}
+	user := DiscordUserData{}
+
+	if err := json.NewDecoder(res.Body).Decode(&user); err != nil {
+		return nil, err
+	}
+	log.Println(user)
```
Comment (jveski) [inline discord/discord.go]: Should either add some context (e.g. `log.Printf("discord user: %v", user)`) or remove this

**Review by jveski — APPROVED** [review]

File: keycloak/keycloak.go (hunk 1 above)
Comment (ATechAdventurer) [reply keycloak/keycloak.go]: Good to know, will fix

File: discord/discord.go (hunk 3 above)
Comment (ATechAdventurer) [reply discord/discord.go]: Yea that was a leftover line from debugging

**Review by jveski — APPROVED** [review]

**Pull request merged**